isisd/isis_cli.c

```diff
@@ -1429,9 +1425,43 @@ DEFPY_YANG(isis_redistribute, isis_redistribute_cmd,
 				      metric_str ? metric_str : NULL);
 	}
 
-	return nb_cli_apply_changes(
-		vty, "./redistribute/%s[protocol='%s'][level='%s']", ip, proto,
-		level);
+	return nb_cli_apply_changes(vty,
+				    "./redistribute/%s[protocol='%s'][level='%s']",
+				    ip, proto, level);
+}
+
+/*
+ * XPath: /frr-isisd:isis/instance/route_leanking
+ */
+DEFPY_YANG(isis_leanking, isis_leanking_cmd,
```

Comment:
First, this should be `leaking`, not `leaNking`. This needs to be fixed everywhere in the code.

Second, I think it should be a part of the regular `redistribute` command. For example, that's how you configure this on Cisco:
```
redistribute isis ip level-2 into level-1 route-map r1
```
IMO we should just use the same command.